Add section in docs about private link setup (#11883)
DataDog/integrations-core: snowflake/README.md

**NOTE**: Snowflake automatically formats the proxy configurations and sets [standard proxy environment variables][8]. 
These variables also impact every requests from integrations, including orchestrators like Docker, ECS, and Kubernetes.

#### Private connectivity to Snowflake configuration

> If you have [private connectivity][9], such as [AWS PrivateLink][10], enabled in Snowflake, you can configure the Snowflake integration by updating the `account` configuration option to be in the following format:

Reviewer: ```suggestion
If [private connectivity][9] (such as [AWS PrivateLink][10]) is enabled in Snowflake, you can configure the Snowflake integration by updating the `account` configuration option to the following format:
```
